### PR TITLE
use clone(for_test=True) replace get_inference_program

### PR DESCRIPTION
**Pull request by luotao1**
related https://github.com/PaddlePaddle/paddle-ce-latest-kpis/pull/158

### CODE REVIEW

**Review by panyx0718 — APPROVED** [review]

**Review by panyx0718 — DISMISSED** [review]

File: python/paddle/fluid/tests/unittests/dist_transformer.py [hunk 1]
```diff
@@ -440,9 +440,7 @@ def split_data(data, num_part):
 def test_context(train_progm, avg_cost, train_exe, dev_count, data_input_names,
                  sum_cost, token_num):
     # Context to do validation.
-    test_program = train_progm.clone()
-    with fluid.program_guard(test_program):
-        test_program = fluid.io.get_inference_program([avg_cost])
+    test_program = train_progm.clone(for_test=True)
```
Comment (panyx0718) [inline python/paddle/fluid/tests/unittests/dist_transformer.py]: Does this work? I'm afraid that the train_program contains backward already. It should probably be called before Optimizer

Comment (gongweibao) [reply python/paddle/fluid/tests/unittests/dist_transformer.py]: Maybe there is a bug here. You can make an issue assigned to me.

If there is still a difference between them, we can mv `get_inference_program` into dist_transfomer.py and rename it use `_` prefix.

Comment (luotao1) [reply python/paddle/fluid/tests/unittests/dist_transformer.py]: After move `test_program = train_progm.clone(for_test=True)` before Optimizer, `test_dist_transformer` runs OK.
```
[07:05:52][Step 1/1]         Start 401: test_dist_transformer
[07:10:11][Step 1/1] 424/428 Test #401: test_dist_transformer ...........................   Passed  259.23 sec
```
http://ci.paddlepaddle.org/viewLog.html?buildId=19035&tab=buildLog&buildTypeId=Paddle_PrCi&logTab=tree&filter=all&_focus=18687

File: python/paddle/fluid/transpiler/__init__.py [hunk 2]
```diff
@@ -15,6 +15,7 @@
 from __future__ import print_function
 
 from .distribute_transpiler import DistributeTranspiler, DistributeTranspilerConfig
+from .inference_transpiler import InferenceTranspiler
```
Comment (luotao1) [inline python/paddle/fluid/transpiler/__init__.py]: If remove this line, errors:
```
[06:38:57][Step 1/1]     t = fluid.transpiler.InferenceTranspiler()
[06:38:57][Step 1/1] AttributeError: 'module' object has no attribute 'InferenceTranspiler'
```
http://ci.paddlepaddle.org/viewLog.html?buildId=19035&tab=buildLog&buildTypeId=Paddle_PrCi&logTab=tree&filter=all&_focus=17841

**Review by panyx0718 — APPROVED** [review]

**Review by typhoonzero — APPROVED** [review]

**Pull request merged**